gpflow/conditionals.py

```diff
@@ -16,12 +16,133 @@
 import tensorflow as tf
 
 from . import settings, mean_functions
+from .core.errors import GPflowError
 from .decors import name_scope
-from .features import InducingPoints
+from .dispatch import dispatch
 from .expectations import expectation
+from .features import InducingPoints, InducingFeature
+from .kernels import Kernel
+from .multikernels import MultiKernel, IndependentMultiKernel, IndependentFeature, MultiInducingPoints
 from .probability_distributions import Gaussian
 
 
+# TODO: Make all outputs of conditionals equal
+# TODO: Add tensorflow assertions of shapes
+# TODO: Remove `conditional()`?
+# TODO: Ensure that R is handled correctly in all cases
+# TODO: Should there be consistentcy between fmean out, and f in?
+# Shapes to keep constant:
+#  - f      : M x L x R  or M x L  or  M x R
+#  - q_sqrt :
+
+
+@dispatch(InducingFeature, Kernel, object, object)
+@name_scope()
+def feature_conditional(feat, kern, Xnew, f, *, full_cov=False, full_cov_output=False, q_sqrt=None, white=False):
```

Comment:
How about a decorator that only takes the types for feat and kern, and combines the dispatch() and name_scope()? would make the rest of the code tidier.

Also, just call it conditional()! :-)